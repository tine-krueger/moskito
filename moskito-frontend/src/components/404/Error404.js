import styled from 'styled-components'
import { useHistory } from 'react-router-dom'
import Button from '../Button/Button'

export default function Error404() {
    const history = useHistory()
    return(
        <NotFound>
            <h2>Fehler 404: Ups, hier stimmt was nicht!</h2>
            <p>Diese Seite existiert leider nicht (mehr)!</p>
            <Button onClick={handleClick}>Zur√ºck</Button>
        </NotFound>
    )

    function handleClick() {
        history.goBack()
    }
}


const NotFound = styled.div`
    margin: 6em 2.5em;
`