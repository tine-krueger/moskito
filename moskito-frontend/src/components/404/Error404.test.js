import { render, screen } from '@testing-library/react'
import 'jest-styled-components'
import userEvent from '@testing-library/user-event'
import { createMemoryHistory } from 'history'
import renderWithRouter from '../../testSetup/setupTests'
import Error404 from './Error404'


describe('Button', () => {
    it('renders correctly', () => {
      const { container } = render(<Error404/>)
      expect(container).toMatchSnapshot()
    })

    it('contains a 404 headline and a push-back button', () => {
        const { getByRole } = render(<Error404/>)
        expect(getByRole('button', {name: /Zurück/i})).toBeInTheDocument()
        expect(getByRole('heading', {name: /Fehler 404/i})).toBeInTheDocument()
    })

    it('goes back in hisory', () => {
        const backwardsRoute = '/find-campsite'
        const backwardsHistory = createMemoryHistory({initialEntries: [backwardsRoute]})
        const { history } = renderWithRouter(<Error404/>,  {history: backwardsHistory})
        userEvent.click(screen.getByText(/Zurück/i))
        expect(history.location.pathname).toBe('/find-campsite')
    })

  })