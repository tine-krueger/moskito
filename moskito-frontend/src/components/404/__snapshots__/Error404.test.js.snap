// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c1 {
  padding: 1em 2em;
  font-family: 'Arvo';
  font-size: 1.1rem;
  border-radius: 10px;
  width: auto;
  border: none;
  background: var(--background);
  box-shadow: 2px 4px 8px -4px rgba(0,0,0,0.6),-2px -2px 2px 0px rgba(255,255,255,0.5);
  color: var(--headline);
  text-transform: uppercase;
  margin: 1em auto 3em auto;
}

.c1:hover {
  box-shadow: inset 4px 4px 6px 0 rgba(0,0,0,0.2),inset -3px -4px 6px 0 rgba(255,255,255,0.3);
  outline: none;
}

.c0 {
  margin: 6em 2.5em;
}

<div>
  <div
    class="c0"
  >
    <h2>
      Fehler 404: Ups, hier stimmt was nicht!
    </h2>
    <p>
      Diese Seite existiert leider nicht (mehr)!
    </p>
    <button
      class="c1"
    >
      Zur√ºck
    </button>
  </div>
</div>
`;
